src/toil/batchSystems/slurm.py

```diff
@@ -444,6 +442,37 @@ def prepareSbatch(self,
             if cpu is not None:
                 sbatch_line.append(f'--cpus-per-task={math.ceil(cpu)}')
 
+            if gpus:
+                sbatch_line = sbatch_line[:1] + [f'--gres=gpu:{gpus}'] + sbatch_line[1:]
+
+                lowest_priority_gpus = sorted([i for i in self.boss.partitions if i[1]], key=lambda x: x[4])
```

Comment:
Having a magic `[4]` here suggests to me that the partition representation is not the right one. Maybe it wants to be a `NamedTuple`? What is field 4 and how do I know it's the right field to use here?